### PR TITLE
don't pass down 'max_ver' as keyword to fancylogger.deprecated

### PR DESCRIPTION
**Pull request by boegel**
This should have been part of #1896...

@JensTimmerman please review?

### CODE REVIEW

**Comment by JensTimmerman** [comment]
ok, can you add a test case that would have triggered this?

**Comment by boegel** [comment]
@JensTimmerman good point, adding tests made me realise there was another issue in `log.deprecated`, since `max_ver` can also be passed unnamed via `*args`

please rerereview?

File: easybuild/tools/build_log.py [hunk 1]
```diff
@@ -129,13 +129,16 @@ def deprecated(self, msg, ver, *args, **kwargs):
                     else: version to check against max_ver to determine warning vs exception
         :param max_ver: version threshold for warning vs exception (compared to 'ver')
         """
-        max_ver = kwargs.get('max_ver', None)
+        max_ver = kwargs.pop('max_ver', None)
+        if max_ver is None and args:
+            max_ver = args[0]
+            args = args[1:]
```
Comment (JensTimmerman) [inline easybuild/tools/build_log.py]: you have no way to know that max_ver will be args[0], so please don't do this, if you expect it, 
make the function definition 
`def deprecated(self, msg, ver, max_ver=None,*args, **kwargs):`
`

File: easybuild/tools/build_log.py [hunk 2]
```diff
@@ -129,13 +129,16 @@ def deprecated(self, msg, ver, *args, **kwargs):
...
         if max_ver is None:
             msg += "; see %s for more information" % DEPRECATED_DOC_URL
             kwargs['exception'] = EasyBuildError
             fancylogger.FancyLogger.deprecated(self, msg, str(CURRENT_VERSION), ver, *args, **kwargs)
         else:
-            fancylogger.FancyLogger.deprecated(self, msg, ver, max_ver)
+            fancylogger.FancyLogger.deprecated(self, msg, ver, max_ver, *args, **kwargs)
```
Comment (JensTimmerman) [inline easybuild/tools/build_log.py]: your pr title states:

File: easybuild/tools/build_log.py (hunk 1 above)
Comment (boegel) [inline easybuild/tools/build_log.py]: good point, thanks

File: easybuild/tools/build_log.py (hunk 2 above)
Comment (boegel) [inline easybuild/tools/build_log.py]: not as a keyword, I'll clarify the title

**Comment by JensTimmerman** [comment]
looks good to me

**Comment by boegel** [comment]
Thanks for the review @JensTimmerman!

**Pull request merged**